library/src/main/java/com/getstream/sdk/chat/model/Channel.java

```diff
@@ -440,48 +463,62 @@ public void onError(String errMsg, int errCode) {
+            }
+
+            @Override
+            public void onFailure(Call<ChannelState> call, Throwable t) {
+                callback.onError(t.getLocalizedMessage(), -1);
+            }
+        };
+
+
         client.onSetUserCompleted(
-                new ClientConnectionCallback() {
-                    @Override
-                    public void onSuccess(User user) {
-                        client.getApiService().queryChannel(channel.type, channel.id, client.getApiKey(), client.getUserId(), client.getClientID(), request).enqueue(new Callback<ChannelState>() {
-                            @Override
-                            public void onResponse(Call<ChannelState> call, Response<ChannelState> response) {
-                                Log.i(TAG, "channel query: incoming watchers " + response.body().getWatchers().size());
-                                mergeWithState(response.body());
-
-                                if (channel.config == null)
-                                    channel.config = response.body().getChannel().getConfig();
-                                if (channel.channelState == null)
-                                    channel.channelState = response.body();
-
-                                client.addChannelConfig(type, channel.config);
-
-                                if (request.isWatch()) {
-                                    client.addToActiveChannels(channel);
-                                    initialized = true;
-                                }
-
-                                Log.i(TAG, "channel query: merged watchers " + channel.getChannelState().getWatchers().size());
-                                // offline storage
-
-                                getClient().storage().insertMessagesForChannel(channel, response.body().getMessages());
-
-                                callback.onSuccess(response.body());
-                            }
-
-                            @Override
-                            public void onFailure(Call<ChannelState> call, Throwable t) {
-                                callback.onError(t.getLocalizedMessage(), -1);
-                            }
-                        });
+            new ClientConnectionCallback() {
+                final ChannelQueryRequest queryRequest = request.withData(channel.extraData);
+                @Override
+                public void onSuccess(User user) {
+                    if (id == null) {
+                        // channels created without ID will get it populated by the API
+                        client.getApiService().queryChannel(channel.type, client.getApiKey(), client.getUserId(), client.getClientID(), queryRequest).enqueue(queryChannelCallback);
+                    } else {
+                        client.getApiService().queryChannel(channel.type, channel.id, client.getApiKey(), client.getUserId(), client.getClientID(), queryRequest).enqueue(queryChannelCallback);
```

Comment:
If we don't use `queryRequest` instead of `request`, it makes new issue for loadMore